canvas/models/models.py

```diff
@@ -58,6 +58,10 @@ def has_create_event_permission(self, user):
         course_reg = get_course_registration(user, self)
         return user.is_teacher or course_reg.registration_type == TA or course_reg.registration_type == INSTRUCTOR
 
+    def has_create_challenge_permission(self, user):
+        course_reg = get_course_registration(user, self)
+        return user.is_teacher or course_reg.registration_type in [reg_type[0] for reg_type in REGISTRATION_TYPE]
```

Comment:
what does this do `course_reg.registration_type in [reg_type[0] for reg_type in REGISTRATION_TYPE]`?